gc/keep.go

```diff
@@ -60,7 +64,7 @@ func addProto(ctx context.Context, k Keep, g bs.Getter, p proto.Message) error {
 	)
 
 	for i := 0; i < fields.Len(); i++ {
-		err := addField(ctx, k, g, refl, fields.Get(i))
+		err := forFieldEdges(refl, fields.Get(i), callback)
```

Comment:
The name `forFieldEdges` implies that the function iterates over fields, but in fact it handles only a single field (though it may end up handling multiple fields via recursion). How about `onFieldEdge` or `handleField` or something like that?

Reply:
I'd meant it as "for each edge within this field," but I don't feel strongly -- done.